fd/jacobian.go

```diff
@@ -0,0 +1,185 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fd
+
+import (
+	"runtime"
+	"sync"
+
+	"github.com/gonum/matrix/mat64"
+)
+
+type JacobianSettings struct {
+	Formula     Formula
+	OriginValue []float64
+	Step        float64
+	Concurrent  bool
+}
+
+// Jacobian approximates the Jacobian matrix of a vector-valued function f at
+// the location x. The Jacobian matrix J is the matrix of all first-order
+// partial derivatives of f. If f maps n-dimensional vectors x to m-dimensional
+// vectors y = f(x), J is an m×n matrix given as
+//      ⎡ ∂f_1 / ∂x_1 ... ∂f_1 / ∂x_n ⎤
+//      ⎢      .   .           .      ⎥
+//  J = ⎢      .       .       .      ⎥
+//      ⎢      .           .   .      ⎥
+//      ⎣ ∂f_m / ∂x_1 ... ∂f_m / ∂x_n ⎦
+// or element-wise
```

Comment:
Hmmm, I still prefer the current formulation so unless you strongly object I'll keep it.